eras/shelley-ma/test-suite/src/Test/Cardano/Ledger/ShelleyMA/Serialisation/Roundtrip.hs

```diff
@@ -60,4 +88,129 @@ allEraRoundtripTests =
+    "Coin CBOR roundtrips"
+    [ testCase "decode 0 as Coin for pre-conway" $
+        roundTripCborRangeExpectation minBound (natVersion @8) (Coin (0))
+    , testCase "fail decode 0 as Coin for conway" $
+        roundTripFailureExpectation (natVersion @9) (Coin (0))
+    , testProperty "decode (Negative Int) as Coin for pre-conway" $ \(getNegative -> i) ->
+        roundTripCborRangeExpectation minBound (natVersion @8) (Coin (i))
+    , testProperty "fail decode (Negative Int) as Coin for conway" $ \(getNegative -> i) ->
+        roundTripFailureExpectation (natVersion @9) (Coin i)
+    , testProperty "decode (Positive Int) as Coin for all eras" $ \(getPositive -> i) ->
+        roundTripCborExpectation (Coin i)
+    ]
+
+multiassetCBORTests :: TestTree
+multiassetCBORTests =
+  testGroup
+    "MultiAsset CBOR roundtrips"
+    [ testCase "decode 0 as MultiAsset for pre-conway" $
+        roundTripCborRangeExpectation minBound (natVersion @8) 
+          (MultiAsset (Map.singleton (PolicyID (mkScriptHash @C_Crypto 1)) (Map.singleton (AssetName "asset") 0)))
+    , testCase "fail decode 0 as MultiAsset for conway" $
+        roundTripFailureExpectation (natVersion @9) 
+          (MultiAsset (Map.singleton (PolicyID (mkScriptHash @C_Crypto 1)) (Map.singleton (AssetName "asset") 0)))
+    , testProperty "decode (Negative Int) as MultiAsset for all eras" $ \(getNegative -> i) ->
+        roundTripCborExpectation 
+          (MultiAsset (Map.singleton (PolicyID (mkScriptHash @C_Crypto 1)) (Map.singleton (AssetName "asset") i)))
+    , testProperty "decode (Positive Int) as MultiAsset for all eras" $ \(getPositive -> i) ->
+        roundTripCborExpectation 
+          (MultiAsset (Map.singleton (PolicyID (mkScriptHash @C_Crypto 1)) (Map.singleton (AssetName "asset") i)))
+    ]
+
+maryvalueCBORTests :: TestTree
+maryvalueCBORTests =
+  testGroup
+    "MaryValue CBOR roundtrips"
+    [ testCase "decode 0 as MaryValue for pre-conway" $
+        roundTripCborRangeExpectation minBound (natVersion @8) 
+          (MaryValue 1 (MultiAsset (Map.singleton (PolicyID (mkScriptHash @C_Crypto 1)) (Map.singleton (AssetName "asset") 0))))
+    , testCase "fail decode 0 as MaryValue for conway" $
+        roundTripFailureExpectation (natVersion @9) 
+          (MaryValue 1 (MultiAsset (Map.singleton (PolicyID (mkScriptHash @C_Crypto 1)) (Map.singleton (AssetName "asset") 0))))
+    , testProperty "fail decode (Negative Int) as MaryValue for all eras" $ \(getNegative -> i) ->
+        forM_ [minBound .. maxBound] $ \version ->
+          roundTripFailureExpectation version 
+            (MaryValue 1 (MultiAsset (Map.singleton (PolicyID (mkScriptHash @C_Crypto 1)) (Map.singleton (AssetName "asset") i))))
+    , testProperty "decode (Positive Int) as MaryValue for all eras" $ \(getPositive -> i) ->
+        roundTripCborExpectation 
+          (MaryValue 1 (MultiAsset (Map.singleton (PolicyID (mkScriptHash @C_Crypto 1)) (Map.singleton (AssetName "asset") i))))
```

Comment:
I think it would be better to create a generator for MultiAsset, that could be used in the definition of the Arbitrary instance for MaryValue as well

```suggestion
    , testProperty "decode (Positive Int) as MaryValue for all eras" $ \ (Positive c) ->
      forAll $ genPositiveMultiAsset $ \ma ->  roundTripCborExpectation (MaryValue c ma)
```
Here is an idea for a potential generator  that does the right thing:

```haskell
genMap genKey genVal = ...

genMultiAsset genAmount = MultiAsset <$> genMap arbitrary (genMap arbitrary genAmount)

genMultiAssetMint = genMultiAsset (getPositive <$> arbitrary)
```
```